test/shaping/run-tests.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import sys, os, subprocess
+
+def cmd(command):
+	p = subprocess.Popen (
+		command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+	p.wait ()
+	print (p.stderr.read(), file=sys.stderr)
+	return p.stdout.read ().decode ("utf-8"), p.returncode
+
+srcdir = os.environ.get ("srcdir", ".")
+builddir = os.environ.get ("builddir", ".")
+top_builddir = os.environ.get ("top_builddir", os.path.normpath(os.path.join(os.getcwd(), "..", "..")))
+EXEEXT = os.environ.get ("EXEEXT", "")
+
+extra_options = "--verify"
+hb_shape = top_builddir + "/util/hb-shape" + EXEEXT
```

Comment:
shouldn't you normalize this to `os.path.join(top_builddir, "util", "hb-shape" + EXEXT)`?

Reply:
Thanks! :) Done